src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -140,31 +140,40 @@ public AnalyzerResult Analyze()
 
             // gather initial local or parameter variable info
             GenerateVariableInfoMap(
-                bestEffort: false, model, dataFlowAnalysisData, symbolMap,
-                out var variableInfoMap, out var failedVariables);
+                bestEffort: false, model, dataFlowAnalysisData, symbolMap, out var variableInfoMap, out var failedVariables);
             if (failedVariables.Count > 0)
             {
                 // If we weren't able to figure something out, go back and regenerate the map
                 // this time in 'best effort' mode.  We'll give the user a message saying there
                 // was a problem, but we allow them to proceed so they're not unnecessarily
                 // blocked just because we didn't understand something.
                 GenerateVariableInfoMap(
-                    bestEffort: true, model, dataFlowAnalysisData, symbolMap,
-                    out variableInfoMap, out var unused);
+                    bestEffort: true, model, dataFlowAnalysisData, symbolMap, out variableInfoMap, out var unused);
                 Contract.ThrowIfFalse(unused.Count == 0);
             }
 
             var thisParameterBeingRead = (IParameterSymbol?)dataFlowAnalysisData.ReadInside.FirstOrDefault(IsThisParameter);
             var isThisParameterWritten = dataFlowAnalysisData.WrittenInside.Any(static s => IsThisParameter(s));
 
+            // Need to generate an instance method if any primary constructor parameter is read or written inside the selection.
+            var primaryConstructorParameterReadOrWritten = dataFlowAnalysisData.ReadInside
+                .Concat(dataFlowAnalysisData.WrittenInside)
+                .OfType<IParameterSymbol>()
+                .FirstOrDefault(s => s.IsPrimaryConstructor(this.CancellationToken)) != null;
```

Comment:
>FirstOrDefault [](http://example.com/codeflow?start=17&length=14)

.Any ?

Reply:
now i want to write an analyzer/fixer that finds this pattern and suggests changing it.